src/types/chain.ts

```diff
@@ -37,6 +37,41 @@ export type ChainConstants = {
     ensRegistry?: ChainContract
     ensUniversalResolver?: ChainContract
     multicall3?: ChainContract
+    // L1 OP Stack Contracts
+    l1CrossDomainMessenger?: {
+      [chainId: number]: unknown
+    }
+    l1Erc721Bridge?: {
+      [chainId: number]: unknown
+    }
+    l1StandardBridge?: {
+      [chainId: number]: unknown
+    }
+    l2OutputOracle?: {
+      [chainId: number]: unknown
+    }
+    optimismMintableErc20Factory?: {
+      [chainId: number]: unknown
+    }
+    optimismPortal?: {
+      [chainId: number]: unknown
+    }
+    systemConfig?: {
+      [chainId: number]: unknown
+    }
+    systemDictator?: {
+      [chainId: number]: unknown
+    }
+    // L2 OP Stack Contracts
+    // note these are redeploys and always have the same address
+    l2CrossDomainMessenger?: Address
+    l2StandardBridge?: Address
+    optimismMintableERC20Factory?: Address
+    gasPriceOracle?: Address
+    l1Block?: Address
+    l2ToL1MessagePasser?: Address
+    l2Erc721Bridge?: Address
+    optimismMintableErc721Factory?: Address
```

Comment:
```suggestion
    // L1 OP Stack Contracts
    optimismL1CrossDomainMessenger?: {
      [chainId: number]: unknown
    }
    optimismL1Erc721Bridge?: {
      [chainId: number]: unknown
    }
    optimismL1StandardBridge?: {
      [chainId: number]: unknown
    }
    optimismL2OutputOracle?: {
      [chainId: number]: unknown
    }
    optimismMintableErc20Factory?: {
      [chainId: number]: unknown
    }
    optimismPortal?: {
      [chainId: number]: unknown
    }
    optimismSystemConfig?: {
      [chainId: number]: unknown
    }
    optimismSystemDictator?: {
      [chainId: number]: unknown
    }
    // L2 OP Stack Contracts
    // note these are redeploys and always have the same address
    optimismL2CrossDomainMessenger?: Address
    optimismL2StandardBridge?: Address
    optimismMintableERC20Factory?: Address
    optimismGasPriceOracle?: Address
    optimismL1Block?: Address
    optimismL2ToL1MessagePasser?: Address
    optimismL2Erc721Bridge?: Address
    optimismMintableErc721Factory?: Address
```
I think we should prefix all optimism addresses with optimism or OP.
Should the address types be `Address | undefined`?

Reply:
yes good call